pkg/arvo/sys/vane/lick.hoon

```diff
@@ -0,0 +1,134 @@
+::  %lick
+!:
+!?  164
+::
+=,  lick
+|=  our=ship
+=>  |%
+    +$  move  [p=duct q=(wite note gift)]
+    +$  note  ~                                         ::  out request $->
+    +$  sign  ~
+    ::
+    +$  lick-state
+      $:  %0
+          unix-duct=duct
+          owners=(map name duct)
+      ==
+    ::
+    +$  name   path
+    --
+::
+~%  %lick  ..part  ~
+::
+=|  lick-state
+=*  state  -
+|=  [now=@da eny=@uvJ rof=roof]
+=*  lick-gate  .
+^?
+|%
+::  +register: Create a move to register an agent with vere
+::
+++  register
+  |=  =name
+  ^-  move
+  [unix-duct.state %give [%spin name]]
+::  +disconnect: Create Move to send a disconnect soak to am agent
+::
+++  disconnect
+  |=  =name
+  ^-  move
+  =/  =duct  (~(get by owners) name)
+  [+.duct %give [%soak name %disconnect ~]]
+::  +call: handle a +task:lick request
+::
+++  call 
+  |=  $:  hen=duct
+          dud=(unit goof)
+          wrapped-task=(hobo task)
+      ==
+  ^-  [(list move) _lick-gate]
+  ::
+  =/  =task  ((harden task) wrapped-task)
+  ?+   -.task  [~ lick-gate]
+      %born     :: need to register devices with vere and send disconnect soak
+    :-  %+  weld 
+          (turn ~(tap in ~(key by owners.state)) register) 
+        (turn ~(tap in ~(key by owners.state)) disconnect)
+    lick-gate(unix-duct hen) 
+    ::
+      %spin     :: A gall agent wants to spin a communication line
+    ::  XX Right now the last agent to register a port owns it. Should
+    ::  this change? 
```

Comment:
Can this just be first come first served? As long as the agent is installed (and not suspended?) it retains ownership, but once uninstalled relinquishes it.